mqtt-streaming/src/test/scala/docs/scaladsl/MqttFlowSpec.scala

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package docs.scaladsl
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.stream.alpakka.mqtt.streaming._
+import akka.stream.alpakka.mqtt.streaming.scaladsl.{ActorMqttClientSession, ActorMqttServerSession, Mqtt}
+import akka.stream.scaladsl.{BroadcastHub, Flow, Keep, Sink, Source, Tcp}
+import akka.stream._
+import akka.testkit.TestKit
+import akka.util.ByteString
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures
+
+import scala.concurrent.duration._
+
+class MqttFlowSpec
+    extends TestKit(ActorSystem("MqttFlowSpec"))
+    with WordSpecLike
+    with Matchers
+    with BeforeAndAfterAll
+    with ScalaFutures {
+
+  private implicit val defaultPatience: PatienceConfig = PatienceConfig(timeout = 5.seconds, interval = 100.millis)
+
+  private implicit val mat: Materializer = ActorMaterializer()
+
+  override def afterAll(): Unit =
+    TestKit.shutdownActorSystem(system)
+
+  "mqtt client flow" should {
+    "establish a bidirectional connection and subscribe to a topic" in {
+      val clientId = "flow-spec/flow"
+      val topic = "source-spec/topic1"
+
+      //#create-streaming-flow
+      val settings = MqttSessionSettings()
+      val session = ActorMqttClientSession(settings)
+
+      val connection = Tcp().outgoingConnection("localhost", 1883)
+
+      val mqttFlow: Flow[Command[_], Either[MqttCodec.DecodeError, Event[_]], NotUsed] =
```

Comment:
You could add a type alias for `Either[MqttCodec.DecodeError, Event[_]]` to make this read better.

Reply:
I already have a Java `DecodeErrorOrEvent`, so I'm unsure what I'd call it. I generally prefer not to hide the types also.